runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -52,20 +52,37 @@ public AbstractQueuedStreamView(final CorfuRuntime runtime,
         super(runtime, streamId, QueuedStreamContext::new);
     }
 
+    /**
+     * Update {@code maxResolution} which is the largest address we
+     * have resolved to so next time we can probably fill from the
+     * {@code resolvedQueue} directly.
+     * <p>
+     * The new {@code maxResolution} may be greater than the actual
+     * tail of this stream, but we know that there is nothing left
+     * to resolve up to that address.
+     *
+     * @param context          stream context
+     * @param newMaxResolution new {@code maxResolution} to set
+     */
+    void updateMaxResolution(QueuedStreamContext context, long newMaxResolution) {
+        if (newMaxResolution != Address.MAX && context.maxResolution < newMaxResolution) {
```

Comment:
how that even possible having newMaxResolution more than Long.MAX?
I mean if you know cases when it's possible, then corfu can work, we have to throw an error?

Reply:
Long.MAX has a meaning: up the latest.

Reply:
yep, I understand, maybe let's check if newMaxResolution == Address.MAX then throw an error?
In real life it never happens, so I don't see any reasons to check that condition

Reply:
Why it cannot happen? we use Address.MAX to indicate we need to sync to the end.

Reply:
It's pretty hard to figure out without knowing particular part of code https://github.com/CorfuDB/CorfuDB/blob/be9ba2d4b4ce9ca5d0e99deede1e49db258368a9/runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java#L76

so, ok